priv/templates/nimble_template/.github/workflows/test.yml.eex

```diff
@@ -87,22 +144,254 @@ jobs:
 
       - name: Run codebase check
         run: mix codebase
+    
+  seed_data:
```

Comment:
What is this step for? For what type of tests is this required? Also, now since it does not appear as a dependency to any step, I am not sure what will is its use 🤔  Also isn't global test data an anti-pattern i.e., should we create only the required test data for each test?

Reply:
Hi @olivierobert this is not the test data, this is the `job for testing the seed script`, as the codebase/schema/migration is changed frequently and the developer does not run the `seed script` on development frequently, so with the codebase updated above, the seed script might be failed, that affect into the new people, who just join the project and running the seed, they need to spend time to fix the seed script.

So with that reason, we decided to have the `test for the seed script` - this job is separated with the `unit test`, here we only test the `seed script` to make sure the seed script is able to run, so the new joiner will happy with that.

The cost for this test is really small, just 10 seconds in most of standard projects, that doesn't take too much CI time, but it bring the benefit to the new joiner.

Reply:
This is an example where the developer modified the schema field name but didn't update the seed script @olivierobert.

![image](https://user-images.githubusercontent.com/11751745/153825853-210b8df8-4b6d-4a0c-a42b-f43cd209a2ea.png)

The result is this seed script won't run :D and the new joiner or who needs to re-run the `seed script` will fix that 😂.

But with the `test seed_script` here, we can avoid that case as the CI will be failed and the modifier need to adjust the seed script along with the ecto schema field too ✅

Reply:
Well understood. Indeed, the seeds are not usually executed that often so ensuring these still run is a good idea. In terms of CI time, it is reasonable. The only downside I see is the amount of workflow code needed to test something small 🙈 

I will add a few suggestions to avoid some confusion like I had.